core/lib/types/src/storage_writes_deduplicator.rs

```diff
@@ -167,17 +181,29 @@ impl StorageWritesDeduplicator {
                 &mut self.metrics.repeated_storage_writes
             };
 
+            let total_size = &mut self.metrics.total_updated_values_size;
+
             match item.update_type {
                 UpdateType::Insert => {
-                    self.modified_key_values.remove(&item.key);
+                    let value = self
+                        .modified_key_values
+                        .remove(&item.key)
+                        .unwrap_or_default();
```

Comment:
better to have `unwrap_or_else(|| panic!` like in the function above, since this mapping is never supposed to be empty